### ISSUE TITLE
Applying cached packages with bad absolute paths

### ISSUE THREAD

**Issue by dave-juicelabs** [microsoft/vcpkg#27028]
**Describe the bug**
```
add_library(skia INTERFACE)
add_library(skia::skia ALIAS skia)
set(SKIA_DEP_DBG dl;pthread;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/debug/lib/libfontconfig.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/debug/lib/libfreetyped.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/debug/lib/libexpat.a;GL;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/debug/lib/libz.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/debug/lib/libharfbuzz.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/debug/lib/libharfbuzz-subset.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/debug/lib/libjpeg.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/debug/lib/libturbojpeg.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/debug/lib/libpng16d.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/debug/lib/libwebp.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/debug/lib/libwebpdemux.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/debug/lib/libwebpdecoder.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/debug/lib/libwebpmux.a)
set(SKIA_DEP_REL dl;pthread;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libfontconfig.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libfreetype.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libexpat.a;GL;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libz.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libharfbuzz.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libharfbuzz-subset.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libjpeg.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libturbojpeg.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libpng16.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libwebp.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libwebpdemux.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libwebpdecoder.a;/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libwebpmux.a)
```

In this sample from skia port configured output file skiaConfig.cmake, some of the dependency files have absolute paths from the original build machine. These files are then cached and when pulled back in for a future build still have the original absolute paths which are no longer valid.

I think what is needed is when vcpkg installs files to any location, it needs to find in those files the absolute path and replace it with VCPKG_INSTALLED_DIR. Then when reusing cached files, VCPKG_INSTALLED_DIR will be defined to the current location and all should be good.

**Environment**
- OS: Windows/Linux (but possibly others)

**To Reproduce**
I will work on a reproduction and attach soon.

**Comment by dave-juicelabs** [comment]
Causes build errors like `ninja: error: '/codebuild/output/src474674433/src/build/vcpkg_installed/x64-linux-juice/lib/libfontconfig.a', needed by '<<some build>>', missing and no known rule to make it`

**Comment by FrankXie05** [comment]
At present, the environment variables about vcpkg can only be used in the project files of vcpkg.

**Comment by dg0yt** [comment]
> At present, the environment variables about vcpkg can only be used in the project files of vcpkg.

I fail to see how this is related to the problem.

The skia port doesn't  use the CMake build system. Maybe it went unnoticed that the generated CMake config still needs `vcpkg_cmake_config_fixup`.
(Such issues are meant to be eventually detected by https://github.com/microsoft/vcpkg-tool/pull/172, but first the issue needs to be resolved in all ports...)

**Comment by FrankXie05** [comment]
Sorry, please ignore my previous message. For `skia`, we can't detect the absolute path like [vcpkg_cmake_config_fixup ](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/ports/vcpkg-cmake-config/vcpkg_cmake_config_fixup.md) and replace it, and it seems that we have to find a way to solve this problem for a specific port before perfecting `port-check`. 🧐

**Comment by dave-juicelabs** [comment]
I have an open PR for that specific issue @dg0yt https://github.com/microsoft/vcpkg/pull/26876.

I will try adding vcpkg_cmake_config_fixup for the ports that I see that issue with my project.

**Comment by dave-juicelabs** [comment]
After adding vcpkg_cmake_config_fixup, the skia config has the issues still.
```
set(SKIA_DEP_DBG Ole32.lib;OleAut32.lib;FontSub.lib;User32.lib;Usp10.lib;Gdi32.lib;OpenGL32.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/debug/lib/zlibd.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/debug/lib/harfbuzz.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/debug/lib/harfbuzz-subset.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/debug/lib/jpeg.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/debug/lib/turbojpeg.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/debug/lib/libpng16d.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/debug/lib/webp.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/debug/lib/webpdemux.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/debug/lib/webpdecoder.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/debug/lib/webpmux.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/debug/lib/libexpatdMD.lib)
set(SKIA_DEP_REL Ole32.lib;OleAut32.lib;FontSub.lib;User32.lib;Usp10.lib;Gdi32.lib;OpenGL32.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/lib/zlib.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/lib/harfbuzz.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/lib/harfbuzz-subset.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/lib/jpeg.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/lib/turbojpeg.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/lib/libpng16.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/lib/webp.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/lib/webpdemux.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/lib/webpdecoder.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/lib/webpmux.lib;/D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows-juice/lib/libexpatMD.lib)
```

Note, these values are dumped by gn. I will work around this but hopefully this will provide a little more information for the port-check changes.

**Comment by FrankXie05** [comment]
> It looks like skia is not dumping out the required files for vcpkg_cmake_config_fixup as it outputs only a single config file.
> 
> ```
> CMake Error at D:/Github/juicelabs/juice/build/vcpkg_installed/x64-windows/share/vcpkg-cmake-config/vcpkg_cmake_config_fixup.cmake:81 (message):
> 
>   'D:/Github/juicelabs/juice/vcpkg/packages/skia_x64-windows-juice/debug/share/skia'
>   does not exist.
> Call Stack (most recent call first):
>   buildtrees/versioning_/versions/skia/4bae2c41ea28c4c82101064e65935683196e091b/portfile.cmake:416 (vcpkg_cmake_config_fixup)
>   scripts/ports.cmake:147 (include)
> ```
> 
> I will work around this by modifying the gn output for the libs but hopefully this will be useful for the port-check changes.

Before perfecting the port check, we can only maintain it for a specific port. I think modifying `gn` is a temporary solution.

**Comment by dg0yt** [comment]
> I will work around this by modifying the gn output for the libs but hopefully this will be useful for the port-check changes.

It is probably easier to just call `vcpkg_replace_string` in the portfile, after moving the setting of `_IMPORT_PREFIX` in `skiaConfig.cmake` to the top of the file:
~~~cmake
vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/${PORT}/skiaConfig.cmake" "${CURRENT_INSTALLED_DIR}" [[${_IMPORT_PREFIX}]])
~~~

If `vcpkg_cmake_config_fixup` isn't used, the dependency on port `vcpkg-cmake-config` must be removed from `vcpkg.json`.

**Comment by dg0yt** [comment]
I think the original skia issue was solved with #28007. (#28971 fixes some issues reported after the update.)
